zcncore/smartcontract_executor_mobile.go

```diff
@@ -0,0 +1,31 @@
+//go:build mobile
+// +build mobile
+
+package zcncore
+
+type SmartContractExecutor interface {
+	// ExecuteSmartContract implements wrapper for smart contract function
+	ExecuteSmartContract(address, methodName string, input string, val uint64) error
+}
+
+func (t *Transaction) ExecuteSmartContract(address, methodName string, input string, val uint64) error {
+	err := t.createSmartContractTxn(address, methodName, input, val)
```

Comment:
I'm afraid we can not use the string `input` directly and pass it to the `t.createSmartContractTxn()`, we will need to decode it, otherwise the transaction input data would not work in 0chain. So do 

```go
scData := make(map[string]interface{})
err := json.Unmarshal([]byte(input), &scData)
t.createSmartContractTxn(address, methodName, scData, val)

```

So as it should be done to `TransactionAuth`

Reply:
In mobile apps, they pass json string as input. eg `"{name:\"facuet\"....}"`

Reply:
It's necessary,  string value of input will not work if it's not decoded into concrete type or the general json map. For example, the below two json encode data are totally different. 0chain smart contract executor will not decode string input.

```go
type Foo struct {
	Name string `json:"name"`
}

sd := SmartContractTxnData{
	Name:      "sc_foo",
	InputArgs: &Foo{Name: "foo"}}

d, _ := json.Marshal(&sd)
fmt.Println(string(d))

f := Foo{Name: "foo"}
fd, _ := json.Marshal(f)

sd1 := SmartContractTxnData{
	Name:      "sc_foo",
	InputArgs: string(fd),
}

d2, _ := json.Marshal(&sd1)
fmt.Println(string(d2))
	
	
# output
{"name":"sc_foo","input":{"name":"foo"}}
{"name":"sc_foo","input":"{\"name\":\"foo\"}"}
```

Reply:
fixed